{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "SwissProt_Proteínas_TRMT61A-Parte 4 do Trabalho\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Código Relativamente as features associadas à proteína codificada pelo gene TRMT61A-Código baseado no tutorial do Biopython"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "NP_689520.2 with 0 features\n"
     ]
    }
   ],
   "source": [
    "#rettype=\"fasta\"- Ficheiro em Formato Fasta\n",
    "from Bio import Entrez\n",
    "from Bio import SeqIO\n",
    "Entrez.email = \"joanafigueiredo2014@gmail.com\"\n",
    "with Entrez.efetch(db=\"protein\", rettype=\"fasta\", retmode=\"text\", id=\"NP_689520.2\") as handle:\n",
    "    seq_record = SeqIO.read(handle, \"fasta\")\n",
    "print(\"%s with %i features\" % (seq_record.id, len(seq_record.features)))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "NP_689520.2 with 15 features\n"
     ]
    }
   ],
   "source": [
    "#rettype=\"gb\" - Ficheiro em Formato GenBank\n",
    "from Bio import Entrez\n",
    "from Bio import SeqIO\n",
    "Entrez.email = \"joanafigueiredo2014@gmail.com\"\n",
    "with Entrez.efetch(db=\"protein\", rettype=\"gb\", retmode=\"text\", id=\"NP_689520.2\") as handle:\n",
    "    seq_record = SeqIO.read(handle, \"gb\") #using \"gb\" as an alias for \"genbank\"\n",
    "print(\"%s with %i features\" % (seq_record.id, len(seq_record.features)))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "NP_689520.2 tRNA (adenine(58)-N(1))-methyltransferase catalyti...\n",
      "Sequence length 289, 15 features, from: Homo sapiens (human)\n"
     ]
    }
   ],
   "source": [
    "from Bio import Entrez\n",
    "from Bio import SeqIO\n",
    "Entrez.email = \"joanafigueiredo2014@gmail.com\"\n",
    "with Entrez.efetch(db=\"protein\", rettype=\"gb\", retmode=\"text\",id=\"NP_689520.2\") as handle:\n",
    "    for seq_record in SeqIO.parse(handle, \"gb\"):\n",
    "        print(\"%s %s...\" % (seq_record.id, seq_record.description[:50]))\n",
    "        print(\"Sequence length %i, %i features, from: %s\"\n",
    "            % (len(seq_record), len(seq_record.features), seq_record.annotations[\"source\"]))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "A base de dados Uniprot permite aceder a toda a informação de um conjunto alargado de proteínas. O Código utilizado no tratamento de ficheiros da SwissProt, foi baseado no tutorial do Biopython e em scripts utilizadas na resolução de algumas fichas das aulas práticas feitas por alunos dos anos anteriores.\n",
    "Tendo em conta: UniProtKB - Q96FX7 (TRMT61A_HUMAN)\n",
    "Nota: O grupo teve problemas com o código apresentado em baixo, não reconheceu certos parâmetros!"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Ignoring: DT   02-MAY-2006, integrated into UniProtKB/Swiss-Prot.\n",
      "Ignoring: DT   01-DEC-2001, sequence version 1.\n",
      "Ignoring: DT   11-DEC-2019, entry version 156.\n",
      "Ignoring: GN   Name=TRMT61A; Synonyms=C14orf172, TRM61;\n",
      "Ignoring: OS   Homo sapiens (Human).\n",
      "Ignoring: OC   Eukaryota; Metazoa; Chordata; Craniata; Vertebrata; Euteleostomi; Mammalia;\n",
      "Ignoring: OC   Eutheria; Euarchontoglires; Primates; Haplorrhini; Catarrhini; Hominidae;\n",
      "Ignoring: OC   Homo.\n",
      "Ignoring: OX   NCBI_TaxID=9606;\n",
      "Ignoring: RN   [1]\n",
      "Ignoring: RP   NUCLEOTIDE SEQUENCE [LARGE SCALE GENOMIC DNA].\n",
      "Ignoring: RX   PubMed=12508121; DOI=10.1038/nature01348;\n",
      "Ignoring: RA   Heilig R., Eckenberg R., Petit J.-L., Fonknechten N., Da Silva C.,\n",
      "Ignoring: RA   Cattolico L., Levy M., Barbe V., De Berardinis V., Ureta-Vidal A.,\n",
      "Ignoring: RA   Pelletier E., Vico V., Anthouard V., Rowen L., Madan A., Qin S., Sun H.,\n",
      "Ignoring: RA   Du H., Pepin K., Artiguenave F., Robert C., Cruaud C., Bruels T.,\n",
      "Ignoring: RA   Jaillon O., Friedlander L., Samson G., Brottier P., Cure S., Segurens B.,\n",
      "Ignoring: RA   Aniere F., Samain S., Crespeau H., Abbasi N., Aiach N., Boscus D.,\n",
      "Ignoring: RA   Dickhoff R., Dors M., Dubois I., Friedman C., Gouyvenoux M., James R.,\n",
      "Ignoring: RA   Madan A., Mairey-Estrada B., Mangenot S., Martins N., Menard M., Oztas S.,\n",
      "Ignoring: RA   Ratcliffe A., Shaffer T., Trask B., Vacherie B., Bellemere C., Belser C.,\n",
      "Ignoring: RA   Besnard-Gonnet M., Bartol-Mavel D., Boutard M., Briez-Silla S.,\n",
      "Ignoring: RA   Combette S., Dufosse-Laurent V., Ferron C., Lechaplais C., Louesse C.,\n",
      "Ignoring: RA   Muselet D., Magdelenat G., Pateau E., Petit E., Sirvain-Trukniewicz P.,\n",
      "Ignoring: RA   Trybou A., Vega-Czarny N., Bataille E., Bluet E., Bordelais I., Dubois M.,\n",
      "Ignoring: RA   Dumont C., Guerin T., Haffray S., Hammadi R., Muanga J., Pellouin V.,\n",
      "Ignoring: RA   Robert D., Wunderle E., Gauguet G., Roy A., Sainte-Marthe L., Verdier J.,\n",
      "Ignoring: RA   Verdier-Discala C., Hillier L.W., Fulton L., McPherson J., Matsuda F.,\n",
      "Ignoring: RA   Wilson R., Scarpelli C., Gyapay G., Wincker P., Saurin W., Quetier F.,\n",
      "Ignoring: RA   Waterston R., Hood L., Weissenbach J.;\n",
      "Ignoring: RT   \"The DNA sequence and analysis of human chromosome 14.\";\n",
      "Ignoring: RL   Nature 421:601-607(2003).\n",
      "Ignoring: RN   [2]\n",
      "Ignoring: RP   NUCLEOTIDE SEQUENCE [LARGE SCALE MRNA].\n",
      "Ignoring: RC   TISSUE=Placenta;\n",
      "Ignoring: RX   PubMed=15489334; DOI=10.1101/gr.2596504;\n",
      "Ignoring: RG   The MGC Project Team;\n",
      "Ignoring: RT   \"The status, quality, and expansion of the NIH full-length cDNA project:\n",
      "Ignoring: RT   the Mammalian Gene Collection (MGC).\";\n",
      "Ignoring: RL   Genome Res. 14:2121-2127(2004).\n",
      "Ignoring: RN   [3]\n",
      "Ignoring: RP   NUCLEOTIDE SEQUENCE [LARGE SCALE MRNA] OF 1-282.\n",
      "Ignoring: RC   TISSUE=Testis;\n",
      "Ignoring: RX   PubMed=14702039; DOI=10.1038/ng1285;\n",
      "Ignoring: RA   Ota T., Suzuki Y., Nishikawa T., Otsuki T., Sugiyama T., Irie R.,\n",
      "Ignoring: RA   Wakamatsu A., Hayashi K., Sato H., Nagai K., Kimura K., Makita H.,\n",
      "Ignoring: RA   Sekine M., Obayashi M., Nishi T., Shibahara T., Tanaka T., Ishii S.,\n",
      "Ignoring: RA   Yamamoto J., Saito K., Kawai Y., Isono Y., Nakamura Y., Nagahari K.,\n",
      "Ignoring: RA   Murakami K., Yasuda T., Iwayanagi T., Wagatsuma M., Shiratori A., Sudo H.,\n",
      "Ignoring: RA   Hosoiri T., Kaku Y., Kodaira H., Kondo H., Sugawara M., Takahashi M.,\n",
      "Ignoring: RA   Kanda K., Yokoi T., Furuya T., Kikkawa E., Omura Y., Abe K., Kamihara K.,\n",
      "Ignoring: RA   Katsuta N., Sato K., Tanikawa M., Yamazaki M., Ninomiya K., Ishibashi T.,\n",
      "Ignoring: RA   Yamashita H., Murakawa K., Fujimori K., Tanai H., Kimata M., Watanabe M.,\n",
      "Ignoring: RA   Hiraoka S., Chiba Y., Ishida S., Ono Y., Takiguchi S., Watanabe S.,\n",
      "Ignoring: RA   Yosida M., Hotuta T., Kusano J., Kanehori K., Takahashi-Fujii A., Hara H.,\n",
      "Ignoring: RA   Tanase T.-O., Nomura Y., Togiya S., Komai F., Hara R., Takeuchi K.,\n",
      "Ignoring: RA   Arita M., Imose N., Musashino K., Yuuki H., Oshima A., Sasaki N.,\n",
      "Ignoring: RA   Aotsuka S., Yoshikawa Y., Matsunawa H., Ichihara T., Shiohata N., Sano S.,\n",
      "Ignoring: RA   Moriya S., Momiyama H., Satoh N., Takami S., Terashima Y., Suzuki O.,\n",
      "Ignoring: RA   Nakagawa S., Senoh A., Mizoguchi H., Goto Y., Shimizu F., Wakebe H.,\n",
      "Ignoring: RA   Hishigaki H., Watanabe T., Sugiyama A., Takemoto M., Kawakami B.,\n",
      "Ignoring: RA   Yamazaki M., Watanabe K., Kumagai A., Itakura S., Fukuzumi Y., Fujimori Y.,\n",
      "Ignoring: RA   Komiyama M., Tashiro H., Tanigami A., Fujiwara T., Ono T., Yamada K.,\n",
      "Ignoring: RA   Fujii Y., Ozaki K., Hirao M., Ohmori Y., Kawabata A., Hikiji T.,\n",
      "Ignoring: RA   Kobatake N., Inagaki H., Ikema Y., Okamoto S., Okitani R., Kawakami T.,\n",
      "Ignoring: RA   Noguchi S., Itoh T., Shigeta K., Senba T., Matsumura K., Nakajima Y.,\n",
      "Ignoring: RA   Mizuno T., Morinaga M., Sasaki M., Togashi T., Oyama M., Hata H.,\n",
      "Ignoring: RA   Watanabe M., Komatsu T., Mizushima-Sugano J., Satoh T., Shirai Y.,\n",
      "Ignoring: RA   Takahashi Y., Nakagawa K., Okumura K., Nagase T., Nomura N., Kikuchi H.,\n",
      "Ignoring: RA   Masuho Y., Yamashita R., Nakai K., Yada T., Nakamura Y., Ohara O.,\n",
      "Ignoring: RA   Isogai T., Sugano S.;\n",
      "Ignoring: RT   \"Complete sequencing and characterization of 21,243 full-length human\n",
      "Ignoring: RT   cDNAs.\";\n",
      "Ignoring: RL   Nat. Genet. 36:40-45(2004).\n",
      "Ignoring: RN   [4]\n",
      "Ignoring: RP   FUNCTION, CATALYTIC ACTIVITY, SUBUNIT, AND VARIANT ALA-66.\n",
      "Ignoring: RX   PubMed=16043508; DOI=10.1261/rna.5040605;\n",
      "Ignoring: RA   Ozanick S., Krecic A., Andersland J., Anderson J.T.;\n",
      "Ignoring: RT   \"The bipartite structure of the tRNA m1A58 methyltransferase from S.\n",
      "Ignoring: RT   cerevisiae is conserved in humans.\";\n",
      "Ignoring: RL   RNA 11:1281-1290(2005).\n",
      "Ignoring: RN   [5]\n",
      "Ignoring: RP   PHOSPHORYLATION [LARGE SCALE ANALYSIS] AT SER-263, AND IDENTIFICATION BY\n",
      "Ignoring: RP   MASS SPECTROMETRY [LARGE SCALE ANALYSIS].\n",
      "Ignoring: RC   TISSUE=Leukemic T-cell;\n",
      "Ignoring: RX   PubMed=19690332; DOI=10.1126/scisignal.2000007;\n",
      "Ignoring: RA   Mayya V., Lundgren D.H., Hwang S.-I., Rezaul K., Wu L., Eng J.K.,\n",
      "Ignoring: RA   Rodionov V., Han D.K.;\n",
      "Ignoring: RT   \"Quantitative phosphoproteomic analysis of T cell receptor signaling\n",
      "Ignoring: RT   reveals system-wide modulation of protein-protein interactions.\";\n",
      "Ignoring: RL   Sci. Signal. 2:RA46-RA46(2009).\n",
      "Ignoring: RN   [6]\n",
      "Ignoring: RP   IDENTIFICATION BY MASS SPECTROMETRY [LARGE SCALE ANALYSIS].\n",
      "Ignoring: RX   PubMed=21269460; DOI=10.1186/1752-0509-5-17;\n",
      "Ignoring: RA   Burkard T.R., Planyavsky M., Kaupe I., Breitwieser F.P., Buerckstuemmer T.,\n",
      "Ignoring: RA   Bennett K.L., Superti-Furga G., Colinge J.;\n",
      "Ignoring: RT   \"Initial characterization of the human central proteome.\";\n",
      "Ignoring: RL   BMC Syst. Biol. 5:17-17(2011).\n",
      "Ignoring: RN   [7]\n",
      "Ignoring: RP   ACETYLATION [LARGE SCALE ANALYSIS] AT SER-2, CLEAVAGE OF INITIATOR\n",
      "Ignoring: RP   METHIONINE [LARGE SCALE ANALYSIS], AND IDENTIFICATION BY MASS SPECTROMETRY\n",
      "Ignoring: RP   [LARGE SCALE ANALYSIS].\n",
      "Ignoring: RX   PubMed=22223895; DOI=10.1074/mcp.m111.015131;\n",
      "Ignoring: RA   Bienvenut W.V., Sumpton D., Martinez A., Lilla S., Espagne C., Meinnel T.,\n",
      "Ignoring: RA   Giglione C.;\n",
      "Ignoring: RT   \"Comparative large-scale characterisation of plant vs. mammal proteins\n",
      "Ignoring: RT   reveals similar and idiosyncratic N-alpha acetylation features.\";\n",
      "Ignoring: RL   Mol. Cell. Proteomics 11:M111.015131-M111.015131(2012).\n",
      "Ignoring: RN   [8]\n",
      "Ignoring: RP   FUNCTION, AND CATALYTIC ACTIVITY.\n",
      "Ignoring: RX   PubMed=29107537; DOI=10.1016/j.molcel.2017.10.019;\n",
      "Ignoring: RA   Li X., Xiong X., Zhang M., Wang K., Chen Y., Zhou J., Mao Y., Lv J., Yi D.,\n",
      "Ignoring: RA   Chen X.W., Wang C., Qian S.B., Yi C.;\n",
      "Ignoring: RT   \"Base-resolution mapping reveals distinct m1A methylome in nuclear- and\n",
      "Ignoring: RT   mitochondrial-encoded transcripts.\";\n",
      "Ignoring: RL   Mol. Cell 0:0-0(2017).\n",
      "Ignoring: RN   [9]\n",
      "Ignoring: RP   FUNCTION, AND CATALYTIC ACTIVITY.\n",
      "Ignoring: RX   PubMed=29072297; DOI=10.1038/nature24456;\n",
      "Ignoring: RA   Safra M., Sas-Chen A., Nir R., Winkler R., Nachshon A., Bar-Yaacov D.,\n",
      "Ignoring: RA   Erlacher M., Rossmanith W., Stern-Ginossar N., Schwartz S.;\n",
      "Ignoring: RT   \"The m(1)A landscape on cytosolic and mitochondrial mRNA at single-base\n",
      "Ignoring: RT   resolution.\";\n",
      "Ignoring: RL   Nature 551:251-255(2017).\n",
      "Ignoring: RN   [10]\n",
      "Ignoring: RP   X-RAY CRYSTALLOGRAPHY (2.00 ANGSTROMS) IN COMPLEX WITH TRMT6;\n",
      "Ignoring: RP   S-ADENOSYL-L-METHIONINE AND TRNA SUBSTRATE, AND SUBUNIT.\n",
      "Ignoring: RX   PubMed=26470919; DOI=10.1016/j.jmb.2015.10.005;\n",
      "Ignoring: RA   Finer-Moore J., Czudnochowski N., O'Connell J.D. III, Wang A.L.,\n",
      "Ignoring: RA   Stroud R.M.;\n",
      "Ignoring: RT   \"Crystal structure of the human tRNA m(1)A58 methyltransferase-tRNA(3)(Lys)\n",
      "Ignoring: RT   complex: refolding of substrate tRNA allows access to the methylation\n",
      "Ignoring: RT   target.\";\n",
      "Ignoring: RL   J. Mol. Biol. 427:3862-3876(2015).\n",
      "Ignoring: CC   -!- FUNCTION: Catalytic subunit of tRNA (adenine-N(1)-)-methyltransferase,\n",
      "Ignoring: CC       which catalyzes the formation of N(1)-methyladenine at position 58\n",
      "Ignoring: CC       (m1A58) in initiator methionyl-tRNA (PubMed:16043508). Catalytic\n",
      "Ignoring: CC       subunit of mRNA N(1)-methyltransferase complex, which mediates\n",
      "Ignoring: CC       methylation of adenosine residues at the N(1) position of a small\n",
      "Ignoring: CC       subset of mRNAs: N(1) methylation takes place in tRNA T-loop-like\n",
      "Ignoring: CC       structures of mRNAs and is only present at low stoichiometries\n",
      "Ignoring: CC       (PubMed:29107537, PubMed:29072297). {ECO:0000269|PubMed:16043508,\n",
      "Ignoring: CC       ECO:0000269|PubMed:29072297, ECO:0000269|PubMed:29107537}.\n",
      "Ignoring: CC   -!- CATALYTIC ACTIVITY:\n",
      "Ignoring: CC       Reaction=adenosine(58) in tRNA + S-adenosyl-L-methionine = H(+) + N(1)-\n",
      "Ignoring: CC         methyladenosine(58) in tRNA + S-adenosyl-L-homocysteine;\n",
      "Ignoring: CC         Xref=Rhea:RHEA:43152, Rhea:RHEA-COMP:10365, Rhea:RHEA-COMP:10366,\n",
      "Ignoring: CC         ChEBI:CHEBI:15378, ChEBI:CHEBI:57856, ChEBI:CHEBI:59789,\n",
      "Ignoring: CC         ChEBI:CHEBI:74411, ChEBI:CHEBI:74491; EC=2.1.1.220;\n",
      "Ignoring: CC         Evidence={ECO:0000255|PROSITE-ProRule:PRU00952,\n",
      "Ignoring: CC         ECO:0000269|PubMed:16043508};\n",
      "Ignoring: CC   -!- CATALYTIC ACTIVITY:\n",
      "Ignoring: CC       Reaction=adenosine in mRNA + S-adenosyl-L-methionine = H(+) + N(1)-\n",
      "Ignoring: CC         methyladenosine in mRNA + S-adenosyl-L-homocysteine;\n",
      "Ignoring: CC         Xref=Rhea:RHEA:55392, Rhea:RHEA-COMP:12414, Rhea:RHEA-COMP:12415,\n",
      "Ignoring: CC         ChEBI:CHEBI:15378, ChEBI:CHEBI:57856, ChEBI:CHEBI:59789,\n",
      "Ignoring: CC         ChEBI:CHEBI:74411, ChEBI:CHEBI:74491;\n",
      "Ignoring: CC         Evidence={ECO:0000269|PubMed:29072297, ECO:0000269|PubMed:29107537};\n",
      "Ignoring: CC   -!- SUBUNIT: Heterotetramer; composed of two copies of TRMT6 and two copies\n",
      "Ignoring: CC       of TRMT61A. {ECO:0000269|PubMed:26470919, ECO:0000305|PubMed:16043508}.\n",
      "Ignoring: CC   -!- INTERACTION:\n",
      "Ignoring: CC       Q9UJA5:TRMT6; NbExp=2; IntAct=EBI-934042, EBI-934061;\n",
      "Ignoring: CC   -!- SUBCELLULAR LOCATION: Nucleus {ECO:0000250|UniProtKB:P46959}.\n",
      "Ignoring: CC   -!- SIMILARITY: Belongs to the class I-like SAM-binding methyltransferase\n",
      "Ignoring: CC       superfamily. TRM61 family. {ECO:0000255|PROSITE-ProRule:PRU00952}.\n",
      "Ignoring: CC   -!- SEQUENCE CAUTION:\n",
      "Ignoring: CC       Sequence=BAC05168.1; Type=Miscellaneous discrepancy; Note=Probable cloning artifact.; Evidence={ECO:0000305};\n",
      "Ignoring: CC   ---------------------------------------------------------------------------\n",
      "Ignoring: CC   Copyrighted by the UniProt Consortium, see https://www.uniprot.org/terms\n",
      "Ignoring: CC   Distributed under the Creative Commons Attribution (CC BY 4.0) License\n",
      "Ignoring: CC   ---------------------------------------------------------------------------\n",
      "Ignoring: DR   EMBL; AL133367; -; NOT_ANNOTATED_CDS; Genomic_DNA.\n",
      "Ignoring: DR   EMBL; AL139300; -; NOT_ANNOTATED_CDS; Genomic_DNA.\n",
      "Ignoring: DR   EMBL; BC010167; AAH10167.1; -; mRNA.\n",
      "Ignoring: DR   EMBL; AK097771; BAC05168.1; ALT_SEQ; mRNA.\n",
      "Ignoring: DR   CCDS; CCDS41994.1; -.\n",
      "Ignoring: DR   RefSeq; NP_689520.2; NM_152307.2.\n",
      "Ignoring: DR   PDB; 5CCB; X-ray; 2.00 A; A=1-289.\n",
      "Ignoring: DR   PDB; 5CCX; X-ray; 2.10 A; A=1-289.\n",
      "Ignoring: DR   PDB; 5CD1; X-ray; 3.60 A; A/D=1-289.\n",
      "Ignoring: DR   PDBsum; 5CCB; -.\n",
      "Ignoring: DR   PDBsum; 5CCX; -.\n",
      "Ignoring: DR   PDBsum; 5CD1; -.\n",
      "Ignoring: DR   SMR; Q96FX7; -.\n",
      "Ignoring: DR   BioGrid; 125450; 26.\n",
      "Ignoring: DR   IntAct; Q96FX7; 17.\n",
      "Ignoring: DR   MINT; Q96FX7; -.\n",
      "Ignoring: DR   STRING; 9606.ENSP00000374399; -.\n",
      "Ignoring: DR   iPTMnet; Q96FX7; -.\n",
      "Ignoring: DR   PhosphoSitePlus; Q96FX7; -.\n",
      "Ignoring: DR   BioMuta; TRMT61A; -.\n",
      "Ignoring: DR   DMDM; 74760827; -.\n",
      "Ignoring: DR   EPD; Q96FX7; -.\n",
      "Ignoring: DR   jPOST; Q96FX7; -.\n",
      "Ignoring: DR   MassIVE; Q96FX7; -.\n",
      "Ignoring: DR   MaxQB; Q96FX7; -.\n",
      "Ignoring: DR   PaxDb; Q96FX7; -.\n",
      "Ignoring: DR   PeptideAtlas; Q96FX7; -.\n",
      "Ignoring: DR   PRIDE; Q96FX7; -.\n",
      "Ignoring: DR   ProteomicsDB; 76570; -.\n",
      "Ignoring: DR   DNASU; 115708; -.\n",
      "Ignoring: DR   Ensembl; ENST00000389749; ENSP00000374399; ENSG00000166166.\n",
      "Ignoring: DR   GeneID; 115708; -.\n",
      "Ignoring: DR   KEGG; hsa:115708; -.\n",
      "Ignoring: DR   UCSC; uc001yng.4; human.\n",
      "Ignoring: DR   CTD; 115708; -.\n",
      "Ignoring: DR   DisGeNET; 115708; -.\n",
      "Ignoring: DR   EuPathDB; HostDB:ENSG00000166166.12; -.\n",
      "Ignoring: DR   GeneCards; TRMT61A; -.\n",
      "Ignoring: DR   HGNC; HGNC:23790; TRMT61A.\n",
      "Ignoring: DR   neXtProt; NX_Q96FX7; -.\n",
      "Ignoring: DR   OpenTargets; ENSG00000166166; -.\n",
      "Ignoring: DR   PharmGKB; PA164726783; -.\n",
      "Ignoring: DR   eggNOG; KOG2915; Eukaryota.\n",
      "Ignoring: DR   eggNOG; COG2519; LUCA.\n",
      "Ignoring: DR   GeneTree; ENSGT00940000154239; -.\n",
      "Ignoring: DR   HOGENOM; HOG000173282; -.\n",
      "Ignoring: DR   InParanoid; Q96FX7; -.\n",
      "Ignoring: DR   KO; K07442; -.\n",
      "Ignoring: DR   OMA; VVYPKDA; -.\n",
      "Ignoring: DR   OrthoDB; 1032689at2759; -.\n",
      "Ignoring: DR   PhylomeDB; Q96FX7; -.\n",
      "Ignoring: DR   TreeFam; TF315053; -.\n",
      "Ignoring: DR   BRENDA; 2.1.1.220; 2681.\n",
      "Ignoring: DR   Reactome; R-HSA-6782315; tRNA modification in the nucleus and cytosol.\n",
      "Ignoring: DR   ChiTaRS; TRMT61A; human.\n",
      "Ignoring: DR   GenomeRNAi; 115708; -.\n",
      "Ignoring: DR   Pharos; Q96FX7; Tdark.\n",
      "Ignoring: DR   PRO; PR:Q96FX7; -.\n",
      "Ignoring: DR   Proteomes; UP000005640; Chromosome 14.\n",
      "Ignoring: DR   RNAct; Q96FX7; protein.\n",
      "Ignoring: DR   Bgee; ENSG00000166166; Expressed in 232 organ(s), highest expression level in epithelium of bronchus.\n",
      "Ignoring: DR   ExpressionAtlas; Q96FX7; baseline and differential.\n",
      "Ignoring: DR   Genevisible; Q96FX7; HS.\n",
      "Ignoring: DR   GO; GO:0005654; C:nucleoplasm; TAS:Reactome.\n",
      "Ignoring: DR   GO; GO:0005634; C:nucleus; IBA:GO_Central.\n",
      "Ignoring: DR   GO; GO:0031515; C:tRNA (m1A) methyltransferase complex; IBA:GO_Central.\n",
      "Ignoring: DR   GO; GO:0061953; F:mRNA (adenine-N1-)-methyltransferase activity; IDA:UniProtKB.\n",
      "Ignoring: DR   GO; GO:0016429; F:tRNA (adenine-N1-)-methyltransferase activity; IEA:InterPro.\n",
      "Ignoring: DR   GO; GO:0080009; P:mRNA methylation; IDA:UniProtKB.\n",
      "Ignoring: DR   GO; GO:0030488; P:tRNA methylation; IBA:GO_Central.\n",
      "Ignoring: DR   GO; GO:0006400; P:tRNA modification; TAS:Reactome.\n",
      "Ignoring: DR   InterPro; IPR029063; SAM-dependent_MTases.\n",
      "Ignoring: DR   InterPro; IPR014816; tRNA_MeTrfase_Gcd14.\n",
      "Ignoring: DR   PANTHER; PTHR12133; PTHR12133; 1.\n",
      "Ignoring: DR   Pfam; PF08704; GCD14; 1.\n",
      "Ignoring: DR   PIRSF; PIRSF017269; GCD14; 1.\n",
      "Ignoring: DR   SUPFAM; SSF53335; SSF53335; 1.\n",
      "Ignoring: DR   PROSITE; PS51620; SAM_TRM61; 1.\n",
      "Ignoring: PE   1: Evidence at protein level;\n",
      "Ignoring: KW   3D-structure; Acetylation; Methyltransferase; Nucleus; Phosphoprotein;\n",
      "Ignoring: KW   Polymorphism; Reference proteome; S-adenosyl-L-methionine; Transferase;\n",
      "Ignoring: KW   tRNA processing.\n",
      "Ignoring: FT   INIT_MET        1\n",
      "Ignoring: FT                   /note=\"Removed\"\n",
      "Ignoring: FT                   /evidence=\"ECO:0000244|PubMed:22223895\"\n",
      "Ignoring: FT   CHAIN           2..289\n",
      "Ignoring: FT                   /note=\"tRNA (adenine(58)-N(1))-methyltransferase catalytic\n",
      "Ignoring: FT                   subunit TRMT61A\"\n",
      "Ignoring: FT                   /id=\"PRO_0000233094\"\n",
      "Ignoring: FT   REGION          20..22\n",
      "Ignoring: FT                   /note=\"Substrate binding\"\n",
      "Ignoring: FT                   /evidence=\"ECO:0000244|PDB:5CD1,\n",
      "Ignoring: FT                   ECO:0000269|PubMed:26470919\"\n",
      "Ignoring: FT   REGION          35..42\n",
      "Ignoring: FT                   /note=\"Substrate binding\"\n",
      "Ignoring: FT                   /evidence=\"ECO:0000244|PDB:5CD1,\n",
      "Ignoring: FT                   ECO:0000269|PubMed:26470919\"\n",
      "Ignoring: FT   REGION          64..65\n",
      "Ignoring: FT                   /note=\"Substrate binding\"\n",
      "Ignoring: FT                   /evidence=\"ECO:0000244|PDB:5CD1,\n",
      "Ignoring: FT                   ECO:0000269|PubMed:26470919\"\n",
      "Ignoring: FT   REGION          85..89\n",
      "Ignoring: FT                   /note=\"Substrate binding\"\n",
      "Ignoring: FT                   /evidence=\"ECO:0000244|PDB:5CD1,\n",
      "Ignoring: FT                   ECO:0000269|PubMed:26470919\"\n",
      "Ignoring: FT   REGION          110..117\n",
      "Ignoring: FT                   /note=\"Substrate binding\"\n",
      "Ignoring: FT                   /evidence=\"ECO:0000244|PDB:5CD1,\n",
      "Ignoring: FT                   ECO:0000269|PubMed:26470919\"\n",
      "Ignoring: FT   REGION          114..116\n",
      "Ignoring: FT                   /note=\"S-adenosyl-L-methionine binding\"\n",
      "Ignoring: FT                   /evidence=\"ECO:0000244|PDB:5CCB, ECO:0000244|PDB:5CCX,\n",
      "Ignoring: FT                   ECO:0000244|PDB:5CD1, ECO:0000269|PubMed:26470919\"\n",
      "Ignoring: FT   REGION          163..164\n",
      "Ignoring: FT                   /note=\"S-adenosyl-L-methionine binding\"\n",
      "Ignoring: FT                   /evidence=\"ECO:0000244|PDB:5CCB, ECO:0000244|PDB:5CCX,\n",
      "Ignoring: FT                   ECO:0000244|PDB:5CD1, ECO:0000269|PubMed:26470919\"\n",
      "Ignoring: FT   REGION          180..183\n",
      "Ignoring: FT                   /note=\"Substrate binding\"\n",
      "Ignoring: FT                   /evidence=\"ECO:0000244|PDB:5CD1,\n",
      "Ignoring: FT                   ECO:0000269|PubMed:26470919\"\n",
      "Ignoring: FT   REGION          205..212\n",
      "Ignoring: FT                   /note=\"Substrate binding\"\n",
      "Ignoring: FT                   /evidence=\"ECO:0000244|PDB:5CD1,\n",
      "Ignoring: FT                   ECO:0000269|PubMed:26470919\"\n",
      "Ignoring: FT   BINDING         87\n",
      "Ignoring: FT                   /note=\"S-adenosyl-L-methionine; via amide nitrogen\"\n",
      "Ignoring: FT                   /evidence=\"ECO:0000244|PDB:5CCB, ECO:0000244|PDB:5CCX,\n",
      "Ignoring: FT                   ECO:0000244|PDB:5CD1, ECO:0000269|PubMed:26470919\"\n",
      "Ignoring: FT   BINDING         135\n",
      "Ignoring: FT                   /note=\"S-adenosyl-L-methionine\"\n",
      "Ignoring: FT                   /evidence=\"ECO:0000244|PDB:5CCB, ECO:0000244|PDB:5CCX,\n",
      "Ignoring: FT                   ECO:0000244|PDB:5CD1, ECO:0000255|PROSITE-ProRule:PRU00952,\n",
      "Ignoring: FT                   ECO:0000269|PubMed:26470919\"\n",
      "Ignoring: FT   BINDING         140\n",
      "Ignoring: FT                   /note=\"S-adenosyl-L-methionine\"\n",
      "Ignoring: FT                   /evidence=\"ECO:0000244|PDB:5CCB, ECO:0000244|PDB:5CCX,\n",
      "Ignoring: FT                   ECO:0000244|PDB:5CD1, ECO:0000269|PubMed:26470919\"\n",
      "Ignoring: FT   BINDING         181\n",
      "Ignoring: FT                   /note=\"S-adenosyl-L-methionine\"\n",
      "Ignoring: FT                   /evidence=\"ECO:0000244|PDB:5CCB, ECO:0000244|PDB:5CCX,\n",
      "Ignoring: FT                   ECO:0000244|PDB:5CD1, ECO:0000255|PROSITE-ProRule:PRU00952,\n",
      "Ignoring: FT                   ECO:0000269|PubMed:26470919\"\n",
      "Ignoring: FT   BINDING         278\n",
      "Ignoring: FT                   /note=\"Substrate\"\n",
      "Ignoring: FT                   /evidence=\"ECO:0000244|PDB:5CD1,\n",
      "Ignoring: FT                   ECO:0000269|PubMed:26470919\"\n",
      "Ignoring: FT   MOD_RES         2\n",
      "Ignoring: FT                   /note=\"N-acetylserine\"\n",
      "Ignoring: FT                   /evidence=\"ECO:0000244|PubMed:22223895\"\n",
      "Ignoring: FT   MOD_RES         263\n",
      "Ignoring: FT                   /note=\"Phosphoserine\"\n",
      "Ignoring: FT                   /evidence=\"ECO:0000244|PubMed:19690332\"\n",
      "Ignoring: FT   VARIANT         66\n",
      "Ignoring: FT                   /note=\"V -> A\"\n",
      "Ignoring: FT                   /evidence=\"ECO:0000269|PubMed:16043508\"\n",
      "Ignoring: FT                   /id=\"VAR_026053\"\n",
      "Ignoring: FT   STRAND          3..5\n",
      "Ignoring: FT                   /evidence=\"ECO:0000244|PDB:5CCB\"\n",
      "Ignoring: FT   STRAND          15..19\n",
      "Ignoring: FT                   /evidence=\"ECO:0000244|PDB:5CCB\"\n",
      "Ignoring: FT   STRAND          25..29\n",
      "Ignoring: FT                   /evidence=\"ECO:0000244|PDB:5CCB\"\n",
      "Ignoring: FT   STRAND          35..38\n",
      "Ignoring: FT                   /evidence=\"ECO:0000244|PDB:5CCB\"\n",
      "Ignoring: FT   STRAND          41..44\n",
      "Ignoring: FT                   /evidence=\"ECO:0000244|PDB:5CCB\"\n",
      "Ignoring: FT   HELIX           45..48\n",
      "Ignoring: FT                   /evidence=\"ECO:0000244|PDB:5CCB\"\n",
      "Ignoring: FT   TURN            49..51\n",
      "Ignoring: FT                   /evidence=\"ECO:0000244|PDB:5CCB\"\n",
      "Ignoring: FT   STRAND          57..59\n",
      "Ignoring: FT                   /evidence=\"ECO:0000244|PDB:5CCB\"\n",
      "Ignoring: FT   STRAND          61..63\n",
      "Ignoring: FT                   /evidence=\"ECO:0000244|PDB:5CCB\"\n",
      "Ignoring: FT   STRAND          65..69\n",
      "Ignoring: FT                   /evidence=\"ECO:0000244|PDB:5CCB\"\n",
      "Ignoring: FT   HELIX           73..77\n",
      "Ignoring: FT                   /evidence=\"ECO:0000244|PDB:5CCB\"\n",
      "Ignoring: FT   HELIX           89..98\n",
      "Ignoring: FT                   /evidence=\"ECO:0000244|PDB:5CCB\"\n",
      "Ignoring: FT   STRAND          106..110\n",
      "Ignoring: FT                   /evidence=\"ECO:0000244|PDB:5CCB\"\n",
      "Ignoring: FT   HELIX           116..125\n",
      "Ignoring: FT                   /evidence=\"ECO:0000244|PDB:5CCB\"\n",
      "Ignoring: FT   TURN            126..128\n",
      "Ignoring: FT                   /evidence=\"ECO:0000244|PDB:5CCB\"\n",
      "Ignoring: FT   STRAND          130..134\n",
      "Ignoring: FT                   /evidence=\"ECO:0000244|PDB:5CCB\"\n",
      "Ignoring: FT   HELIX           138..150\n",
      "Ignoring: FT                   /evidence=\"ECO:0000244|PDB:5CCB\"\n",
      "Ignoring: FT   HELIX           154..156\n",
      "Ignoring: FT                   /evidence=\"ECO:0000244|PDB:5CCB\"\n",
      "Ignoring: FT   STRAND          157..160\n",
      "Ignoring: FT                   /evidence=\"ECO:0000244|PDB:5CCB\"\n",
      "Ignoring: FT   HELIX           164..167\n",
      "Ignoring: FT                   /evidence=\"ECO:0000244|PDB:5CCB\"\n",
      "Ignoring: FT   STRAND          175..180\n",
      "Ignoring: FT                   /evidence=\"ECO:0000244|PDB:5CCB\"\n",
      "Ignoring: FT   HELIX           185..188\n",
      "Ignoring: FT                   /evidence=\"ECO:0000244|PDB:5CCB\"\n",
      "Ignoring: FT   HELIX           189..195\n",
      "Ignoring: FT                   /evidence=\"ECO:0000244|PDB:5CCB\"\n",
      "Ignoring: FT   STRAND          201..209\n",
      "Ignoring: FT                   /evidence=\"ECO:0000244|PDB:5CCB\"\n",
      "Ignoring: FT   HELIX           210..223\n",
      "Ignoring: FT                   /evidence=\"ECO:0000244|PDB:5CCB\"\n",
      "Ignoring: FT   STRAND          226..243\n",
      "Ignoring: FT                   /evidence=\"ECO:0000244|PDB:5CCB\"\n",
      "Ignoring: FT   STRAND          265..273\n",
      "Ignoring: FT                   /evidence=\"ECO:0000244|PDB:5CCB\"\n",
      "Ignoring: FT   STRAND          280..286\n",
      "Ignoring: FT                   /evidence=\"ECO:0000244|PDB:5CCB\"\n",
      "Ignoring: SQ   SEQUENCE   289 AA;  31382 MW;  5DCD6437A0B6F54D CRC64;\n",
      "Ignoring: MSFVAYEELI KEGDTAILSL GHGAMVAVRV QRGAQTQTRH GVLRHSVDLI GRPFGSKVTC\n",
      "Ignoring: GRGGWVYVLH PTPELWTLNL PHRTQILYST DIALITMMLE LRPGSVVCES GTGSGSVSHA\n",
      "Ignoring: IIRTIAPTGH LHTVEFHQQR AEKAREEFQE HRVGRWVTVR TQDVCRSGFG VSHVADAVFL\n",
      "Ignoring: DIPSPWEAVG HAWDALKVEG GRFCSFSPCI EQVQRTCQAL AARGFSELST LEVLPQVYNV\n",
      "Ignoring: RTVSLPPPDL GTGTDGPAGS DTSPFRSGTP MKEAVGHTGY LTFATKTPG\n",
      "TRM61_HUMAN             Reviewed;         289 AA.\n",
      "RecName: Full=tRNA (adenine(58)-N(1))-methyltransferase catalytic subunit TRMT61A; EC=2.1.1.220 {ECO:0000269|PubMed:16043508}; AltName: Full=mRNA methyladenosine-N(1)-methyltransferase catalytic subunit TRMT61A {ECO:0000303|PubMed:29072297}; EC=2.1.1.- {ECO:0000269|PubMed:29072297, ECO:0000269|PubMed:29107537}; AltName: Full=tRNA(m1A58)-methyltransferase subunit TRMT61A; Short=tRNA(m1A58)MTase subunit TRMT61A;\n"
     ]
    }
   ],
   "source": [
    "from Bio import SwissProt\n",
    "from Bio.SwissProt import KeyWList\n",
    "handle= open(\"Q96FX7.txt\")\n",
    "records = KeyWList.parse(handle)\n",
    "for record in records:\n",
    "    print(record[\"ID\"])\n",
    "    print(record[\"DE\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "from Bio import ExPASy\n",
    "from Bio import SwissProt\n",
    "from Bio import SeqIO\n",
    "handle = ExPASy.get_sprot_raw(\"Q96FX7\")\n",
    "seq_record = SeqIO.read(handle, \"swiss\")\n",
    "print(seq_record.id)\n",
    "print(seq_record.name)\n",
    "print(seq_record.description)\n",
    "print(repr(seq_record.seq))\n",
    "print(\"Length %i\" % len(seq_record))\n",
    "print(seq_record.annotations[\"keywords\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "from Bio import Entrez\n",
    "Entrez.email = \"joanafigueiredo2014@gmail.com\"\n",
    "\n",
    "import urllib\n",
    "handle = urllib.urlopen(\"https://www.uniprot.org/uniprot/Q96FX7.txt\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "from Bio import Entrez\n",
    "Entrez.email = \"joanafigueiredo2014@gmail.com\"\n",
    "from Bio import ExPASy\n",
    "from Bio import SwissProt\n",
    "\n",
    "\n",
    "handle = ExPASy.get_sprot_raw(\"Q96FX7\")\n",
    "record = SwissProt.read(handle)save_file = open(\"my_swissprot.txt\", \"w\")\n",
    "save_file.write (\"SWISSPROT REGIST\" + \"\\n\")\n",
    "save_file.write (\"\\n\")\n",
    " \n",
    "for ref in record.references:\n",
    "    text1=\"authors:\"+ ref.authors\n",
    "    text2=\"title:\" + ref.title\n",
    "    text4= record.description\n",
    "    text3=\"Classificação dos organismos:\"+ str(record.organism_classification)   \n",
    "    save_file.write(text4+\"\\n\"+text3+\"\\n\"+text2+\"\\n\"+text1)\n",
    "    save_file.write(\"\\n\")\n",
    "    handle.close()    \n",
    " \n",
    "save_file.close()"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
